// @larksuiteoapi/core/dist/constants/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CallbackType = exports.Domain = exports.AppType = exports.HTTPKeyStatusCode = exports.HTTPHeaderKeyLogID = exports.HTTPHeaderKeyRequestID = exports.HTTPHeaderKey = exports.DefaultContentType = exports.ContentTypeJson = exports.ContentType = void 0;
exports.ContentType = "Content-Type";
exports.ContentTypeJson = "application/json";
exports.DefaultContentType = exports.ContentTypeJson + "; charset=utf-8";
exports.HTTPHeaderKey = "HTTP-Header";
exports.HTTPHeaderKeyRequestID = "X-Request-Id";
exports.HTTPHeaderKeyLogID = "X-Tt-Logid";
exports.HTTPKeyStatusCode = "http_status_code";
var AppType;
(function (AppType) {
    AppType["ISV"] = "isv";
    AppType["Internal"] = "internal";
})(AppType = exports.AppType || (exports.AppType = {}));
var Domain;
(function (Domain) {
    Domain["FeiShu"] = "https://open.feishu.cn";
    Domain["LarkSuite"] = "https://open.larksuite.com";
})(Domain = exports.Domain || (exports.Domain = {}));
var CallbackType;
(function (CallbackType) {
    CallbackType["Event"] = "event_callback";
    CallbackType["Challenge"] = "url_verification";
})(CallbackType = exports.CallbackType || (exports.CallbackType = {}));
