// @larksuiteoapi/core/dist/log/log.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConsoleLogger = exports.LoggerProxy = exports.LoggerLevel = void 0;
var LoggerLevel;
(function (LoggerLevel) {
    LoggerLevel[LoggerLevel["ERROR"] = 4] = "ERROR";
    LoggerLevel[LoggerLevel["WARN"] = 3] = "WARN";
    LoggerLevel[LoggerLevel["INFO"] = 2] = "INFO";
    LoggerLevel[LoggerLevel["DEBUG"] = 1] = "DEBUG";
})(LoggerLevel = exports.LoggerLevel || (exports.LoggerLevel = {}));
class LoggerProxy {
    constructor(level, logger) {
        this.level = level;
        this.logger = logger;
    }
    debug(...msg) {
        if (this.level <= LoggerLevel.DEBUG) {
            this.logger.debug("[debug]", ...msg);
        }
    }
    info(...msg) {
        if (this.level <= LoggerLevel.INFO) {
            this.logger.info("[info]", ...msg);
        }
    }
    warn(...msg) {
        if (this.level <= LoggerLevel.WARN) {
            this.logger.warn("[warn]", ...msg);
        }
    }
    error(...msg) {
        if (this.level <= LoggerLevel.ERROR) {
            this.logger.error("[error]", ...msg);
        }
    }
}
exports.LoggerProxy = LoggerProxy;
class ConsoleLogger {
    debug(...msg) {
        console.log(...msg);
    }
    info(...msg) {
        console.log(...msg);
    }
    warn(...msg) {
        console.log(...msg);
    }
    error(...msg) {
        console.log(...msg);
    }
}
exports.ConsoleLogger = ConsoleLogger;
